dev/ComboBox/ComboBoxAttachedProperties.cpp

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include <pch.h>
+#include <common.h>
+#include "ComboBoxAttachedProperties.h"
+#include "ComboBoxAttachedProperties.properties.h"
+#include "DispatcherHelper.h"
+#include "Converters.h"
+#include "HashMap.h"
+
+static constexpr auto c_popupName = L"Popup"sv;
+static constexpr auto c_popupBorderName = L"PopupBorder"sv;
+static constexpr auto c_editableTextName = L"EditableText"sv;
+
+using CornerRadiusFilterType = CornerRadiusFilterConverter::FilterType;
+
+// Normal ComboBox and editable ComboBox have different CornerRadius behaviors.
+// Xaml is not lifted yet when we implementing this feature so we don't have access to ComboBox code.
+// Creating this attached property to help us plug in some extra logic without touching the actual ComboBox code.
+void ComboBoxAttachedProperties::OnApplyDynamicCornerRadiusPropertyChanged(
+    const winrt::DependencyObject& sender,
+    const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    static std::map<winrt::ComboBox*, std::vector<winrt::event_token>> comboBoxEventTokenMap;
```

Comment:
@kmahone winrt::weak_ref is not comparable (i'm getting compile errors saying operator '<' not implemented on weak_ref) so i'm using raw pointer here, and I think it should be ok?

Reply:
I haven't looked at the broader scope of the change, but raw pointers are essentially only ok when being passed as *in* parameters to functions. Luckily implementing comparison for weak_ref [should be easy](https://stackoverflow.com/questions/5733254/how-can-i-create-my-own-comparator-for-a-map).

Reply:
No, raw pointers won't work here. When the ComboBox gets destroyed, you will still have a raw pointer pointing to invalid memory. With a raw pointer you have no way of knowing if what you are pointing to is still valid. That is what weak ref gets you.